Add release notes and documentation for subctl upgrade (#1009)
submariner-io/submariner-website: src/content/operations/upgrading/_index.en.md

---
title: "Upgrading"
date: 2023-09-06T14:06:11+02:00
weight: 15
---

Starting with Submariner 0.16, the recommended way to upgrade Submariner is via the [`subctl upgrade`](#automated-upgrade) command.

> This can be used to upgrade clusters to Submariner 0.16.

Reviewer: Maybe we can remove this sentence as this will become stale in the next release.

```suggestion
```

To upgrade older clusters to a version of Submariner before 0.16, follow the [manual upgrade process](#manual-upgrade).

## Automated Upgrade

If your current version of `subctl` (as indicated by `subctl version`) is older than 0.16,
start by [installing the desired version of `subctl`](../deployment/subctl/#install).
If your current version of `subctl` is 0.16 or later, it will upgrade itself during the upgrade process.

Once you have `subctl` 0.16 or later, run it with a kubeconfig pointing to the cluster(s) you wish to upgrade:

```bash
subctl upgrade --kubeconfig /path/to/kubeconfig
```

(The `--kubeconfig` parameter is optional; `subctl` will use any configuration that `kubectl` would find.)

> `subctl upgrade` will start by upgrading `subctl` to the latest released version,

Reviewer: Just curious:

What if the kubeconfig has multiple contexts pointing to the same cluster. Does it run upgrade against all the contexts?

Author: Yes, the end of the sentence was supposed to convey that but I should clarify.

and then upgrade all the Submariner components in accessible clusters to match.

A specific target version can be specified using the `--to` parameter:

```bash
subctl upgrade --to v0.16.0
```

## Manual Upgrade

> To manually uninstall Submariner from a cluster, follow the steps below:

Reviewer 2: ```suggestion
To manually upgrade Submariner cluster, follow the steps below:
```
